### PR TITLE
Exposed python function added .none(false)

### PR DESCRIPTION
**Pull request by divshekhar**
Signed-off-by: divshekhar <imdshekhar@gmail.com>

### CODE REVIEW

**Comment by hodoulp** [comment]
The work is about #1202

**Comment by divshekhar** [comment]
> The work is about #1202

Yes sir

File: src/bindings/python/PyConfig.cpp [hunk 1]
```diff
@@ -65,31 +65,31 @@ void bindPyConfig(py::module & m)
...
         .def_static("CreateFromEnv", &Config::CreateFromEnv)
-        .def_static("CreateFromFile", &Config::CreateFromFile, "fileName"_a)
+        .def_static("CreateFromFile", &Config::CreateFromFile, "fileName"_a.none(false))
         .def_static("CreateFromStream", [](const std::string & str) 
             {
                 std::istringstream is(str);
                 return Config::CreateFromStream(is);
             }, 
-             "str"_a)
+             "str"_a.none(false))
```
Comment (BernardLefebvre) [inline src/bindings/python/PyConfig.cpp:74]: It should not be added when it is not needed

File: src/bindings/python/PyConfig.cpp [hunk 2]
```diff
@@ -65,31 +65,31 @@ void bindPyConfig(py::module & m)
...
                     
         .def("getMajorVersion", &Config::getMajorVersion)
-        .def("setMajorVersion", &Config::setMajorVersion, "major"_a)
+        .def("setMajorVersion", &Config::setMajorVersion, "major"_a.none(false))
```
Comment (BernardLefebvre) [inline src/bindings/python/PyConfig.cpp:77]: not needed

File: src/bindings/python/PyConfig.cpp [hunk 3]
```diff
@@ -65,31 +65,31 @@ void bindPyConfig(py::module & m)
...
         .def("getMinorVersion", &Config::getMinorVersion)
-        .def("setMinorVersion", &Config::setMinorVersion, "minor"_a)
+        .def("setMinorVersion", &Config::setMinorVersion, "minor"_a.none(false))
```
Comment (BernardLefebvre) [inline src/bindings/python/PyConfig.cpp:79]: not needed

File: src/bindings/python/PyConfig.cpp [hunk 4]
```diff
@@ -65,31 +65,31 @@ void bindPyConfig(py::module & m)
...
-        .def("setDescription", &Config::setDescription, "description"_a)
+        .def("setDescription", &Config::setDescription, "description"_a.none(false))
         .def("serialize", [](ConfigRcPtr & self, const std::string & fileName) 
             {
                 std::ofstream f(fileName.c_str());
                 self->serialize(f);
                 f.close();
             }, 
-             "fileName"_a)
+             "fileName"_a.none(false))
```
Comment (BernardLefebvre) [inline src/bindings/python/PyConfig.cpp:92]: not needed

File: src/bindings/python/PyConfig.cpp [hunk 5]
```diff
@@ -99,45 +99,45 @@ void bindPyConfig(py::module & m)
         .def("getCacheID", (const char * (Config::*)() const) &Config::getCacheID)
         .def("getCacheID", 
              (const char * (Config::*)(const ConstContextRcPtr &) const) &Config::getCacheID, 
-             "context"_a)
+             "context"_a.none(false))
```
Comment (BernardLefebvre) [inline src/bindings/python/PyConfig.cpp:102]: not needed (because of definition just above)

File: src/bindings/python/PyConfig.cpp [hunk 6]
```diff
@@ -99,45 +99,45 @@ void bindPyConfig(py::module & m)
...
+        .def("addEnvironmentVar", &Config::addEnvironmentVar, "name"_a.none(false), "defaultValue"_a.none(false))
         .def("getEnvironmentVarNames", [](ConfigRcPtr & self) 
             {
                 return EnvironmentVarNameIterator(self);
             })
-        .def("getEnvironmentVarDefault", &Config::getEnvironmentVarDefault, "name"_a)
+        .def("getEnvironmentVarDefault", &Config::getEnvironmentVarDefault, "name"_a.none(false))
         .def("clearEnvironmentVars", &Config::clearEnvironmentVars)
-        .def("setEnvironmentMode", &Config::setEnvironmentMode, "mode"_a)
+        .def("setEnvironmentMode", &Config::setEnvironmentMode, "mode"_a.none(false))
```
Comment (BernardLefebvre) [inline src/bindings/python/PyConfig.cpp:113]: not needed

File: src/bindings/python/PyConfig.cpp [hunk 7]
```diff
@@ -99,45 +99,45 @@ void bindPyConfig(py::module & m)
...
+        .def("getColorSpaces", &Config::getColorSpaces, "category"_a.none(false))
+        .def("getColorSpace", &Config::getColorSpace, "name"_a.none(false))
         .def("getColorSpaceNames", [](ConfigRcPtr & self, 
                                       SearchReferenceSpaceType searchReferenceType, 
                                       ColorSpaceVisibility visibility) 
             {
                 return ColorSpaceNameIterator(self, searchReferenceType, visibility);
             },
-             "searchReferenceType"_a, "visibility"_a)
+             "searchReferenceType"_a.none(false), "visibility"_a.none(false))
```
Comment (BernardLefebvre) [inline src/bindings/python/PyConfig.cpp:133]: not needed

File: src/bindings/python/PyConfig.cpp [hunk 8]
```diff
@@ -99,45 +99,45 @@ void bindPyConfig(py::module & m)
...
         .def("getColorSpaces", [](ConfigRcPtr & self, 
                                   SearchReferenceSpaceType searchReferenceType, 
                                   ColorSpaceVisibility visibility) 
             {
                 return ColorSpaceIterator(self, searchReferenceType, visibility);
             },
-             "searchReferenceType"_a, "visibility"_a)
+             "searchReferenceType"_a.none(false), "visibility"_a.none(false))
```
Comment (BernardLefebvre) [inline src/bindings/python/PyConfig.cpp:140]: not needed

File: src/bindings/python/PyConfig.cpp [hunk 9]
```diff
@@ -231,17 +231,17 @@ void bindPyConfig(py::module & m)
             {
                 self->setDefaultLumaCoefs(rgb.data());
             }, 
-             "rgb"_a)
+             "rgb"_a.none(false))
```
Comment (BernardLefebvre) [inline src/bindings/python/PyConfig.cpp:234]: not needed

File: src/bindings/python/PyConfig.cpp [hunk 10]
```diff
@@ -250,80 +250,80 @@ void bindPyConfig(py::module & m)
...
-             "context"_a, "srcColorSpaceName"_a, "dstColorSpaceName"_a)
+             "context"_a.none(false), "srcColorSpaceName"_a.none(false), "dstColorSpaceName"_a.none(false))
         .def("getProcessor", 
              (ConstProcessorRcPtr (Config::*)(const char *,
                                               const char *,
                                               const char *,
                                               TransformDirection) const) 
              &Config::getProcessor, 
-             "srcColorSpaceName"_a, "display"_a, "view"_a, "direction"_a)
+             "srcColorSpaceName"_a.none(false), "display"_a.none(false), "view"_a.none(false), "direction"_a.none(false))
```
Comment (BernardLefebvre) [inline src/bindings/python/PyConfig.cpp:303]: not needed for direction

File: src/bindings/python/PyConfig.cpp [hunk 11]
```diff
@@ -250,80 +250,80 @@ void bindPyConfig(py::module & m)
...
         .def("getProcessor", 
              (ConstProcessorRcPtr (Config::*)(const ConstContextRcPtr &, 
                                               const char *, 
                                               const char *, 
                                               const char *,
                                               TransformDirection) const) 
              &Config::getProcessor, 
-             "context"_a, "srcColorSpaceName"_a, "display"_a, "view"_a, "direction"_a)
+             "context"_a.none(false), "srcColorSpaceName"_a.none(false), "display"_a.none(false), "view"_a.none(false), "direction"_a.none(false))
```
Comment (BernardLefebvre) [inline src/bindings/python/PyConfig.cpp:311]: not needed for direction

File: src/bindings/python/PyConfig.cpp [hunk 12]
```diff
@@ -250,80 +250,80 @@ void bindPyConfig(py::module & m)
...
              (ConstProcessorRcPtr (Config::*)(const ConstTransformRcPtr &) const) 
              &Config::getProcessor, 
-             "transform"_a)
+             "transform"_a.none(false))
         .def("getProcessor", 
              (ConstProcessorRcPtr (Config::*)(const ConstTransformRcPtr &, 
                                               TransformDirection) const) 
              &Config::getProcessor, 
-             "transform"_a, "direction"_a)
+             "transform"_a.none(false), "direction"_a.none(false))
```
Comment (BernardLefebvre) [inline src/bindings/python/PyConfig.cpp:320]: not needed for direction

File: src/bindings/python/PyConfig.cpp [hunk 13]
```diff
@@ -250,80 +250,80 @@ void bindPyConfig(py::module & m)
...
         .def("getProcessor", 
              (ConstProcessorRcPtr (Config::*)(const ConstContextRcPtr &, 
                                               const ConstTransformRcPtr &, 
                                               TransformDirection) const) 
              &Config::getProcessor, 
-             "context"_a, "transform"_a, "direction"_a)
+             "context"_a.none(false), "transform"_a.none(false), "direction"_a.none(false))
```
Comment (BernardLefebvre) [inline src/bindings/python/PyConfig.cpp:326]: not needed for direction

File: src/bindings/python/PyConfig.cpp [hunk 14]
```diff
@@ -370,10 +370,10 @@ void bindPyConfig(py::module & m)
                 return Config::GetProcessorFromConfigs(srcContext, srcConfig, srcColorSpaceName, srcInterchangeName,
                                                        dstContext, dstConfig, dstColorSpaceName, dstInterchangeName);
             }, 
-                    "srcContext"_a, "srcConfig"_a, "srcColorSpaceName"_a, "srcInterchangeName"_a, 
-                    "dstContext"_a, "dstConfig"_a, "dstColorSpaceName"_a, "dstInterchangeName"_a)
+                    "srcContext"_a.none(false), "srcConfig"_a.none(false), "srcColorSpaceName"_a.none(false), "srcInterchangeName"_a.none(false), 
+                    "dstContext"_a.none(false), "dstConfig"_a.none(false), "dstColorSpaceName"_a.none(false), "dstInterchangeName"_a.none(false))
 
-        .def("setProcessorCacheFlags", &Config::setProcessorCacheFlags, "flags"_a);
+        .def("setProcessorCacheFlags", &Config::setProcessorCacheFlags, "flags"_a.none(false));
```
Comment (BernardLefebvre) [inline src/bindings/python/PyConfig.cpp:376]: not needed

**Comment by BernardLefebvre** [comment]
Please do not change what is already working. You should test things if you are not sure. 

Please add some automated tests (in tests/python/ConfigTest.py)

**Comment by BernardLefebvre** [comment]
Good.
Note that tests are still missing.

**Pull request closed without merging**